### PR TITLE
Dhtarc stability test

### PR DESCRIPTION
**Pull request by freesig**
### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by maackle — APPROVED** [review]

**Comment by maackle** [comment]
This is lots of lines, but 500+ of them are from an ignored test

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 1]
```diff
@@ -1045,7 +1046,8 @@ impl Space {
         let i_s_c = i_s.clone();
         tokio::task::spawn(async move {
             loop {
-                tokio::time::sleep(std::time::Duration::from_secs(5 * 60)).await;
+                // tokio::time::sleep(std::time::Duration::from_secs(5 * 60)).await;
+                tokio::time::sleep(std::time::Duration::from_secs(5)).await;
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:1050]: ^^ Is this an accident from debugging? I don't think we want to be updating the agent info every 5 seconds...

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 2]
```diff
@@ -422,6 +422,7 @@ async fn update_arc_length(
     arc: &mut DhtArc,
 ) -> KitsuneP2pResult<()> {
     let density = evt_sender.query_peer_density(space.clone(), *arc).await?;
+    dbg!(density.expected_count());
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:425]: should this be a trace or removed?

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 3]
```diff
@@ -1234,6 +1236,7 @@ impl Space {
                 .get(agent)
                 .cloned()
                 .unwrap_or_else(|| DhtArc::full(agent.get_loc()))
+            // .unwrap_or_else(|| DhtArc::with_coverage(agent.get_loc(), 0.1))
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:1239]: remove or comment?

File: crates/kitsune_p2p/dht_arc/src/peer_view/peer_view_beta.rs [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+    ///
+    /// The idea here is that you may already have enough peer information to make a
+    /// better extrapolation of the full network in a area more focused around your center
+    /// point. In this case it is not worth waiting for all the peer information to
+    /// sync for the larger view as that might take a long time.
+    ///
+    /// Imagine the case where you join a large network with a full arc, you might have a
+    /// good sample size close to your center long before you sync all the networks peer
+    /// information.
+    fn check_focused_view(&self, arc: DhtArc, peers: &[DhtArc], focus_size: usize) -> Option<f64> {
```
Comment (neonphog) [inline crates/kitsune_p2p/dht_arc/src/peer_view/peer_view_beta.rs:64]: I'm not sure I agree with the philosophy of this function... The only property to me that seems to make for an accurate extrapolation would be completeness of data, and we're making no effort to be more complete in the area closer to us than further away. And if we *do* have complete data for the full area locally, is there any benefit in not using it?

**Review by neonphog — CHANGES_REQUESTED** [review]
Marking as "Request changes" for the 5 second agent info update. I'm fine merging with the 'check_focused_view' thing since it's been tested with it, just want it to be a consideration for any future work.

**Comment by freesig** [comment]
@maackle Can I rebase this onto develop. I'm not sure this diff actually represents what we are wanting to merge to develop?

**Comment by maackle** [comment]
Sure rebase it, but I did merge develop into the feature branch and this
branch a few days ago.

**Review by neonphog — APPROVED** [review]

**Review by maackle — APPROVED** [review]
I still approve

**Pull request merged**